ports/rp2/machine_pin.c

```diff
@@ -240,12 +196,18 @@ STATIC void machine_pin_print(const mp_print_t *print, mp_obj_t self_in, mp_prin
             }
         }
         if (funcsel != GPIO_FUNC_SIO) {
-            mp_printf(print, ", alt=%u", funcsel);
+            const machine_pin_af_obj_t *af = machine_pin_find_af_by_index(self, funcsel);
+            if (af == NULL) {
+                mp_printf(print, ", alt=%u", funcsel);
+            } else {
+                mp_printf(print, ", alt=%q", af->name);
+            }
         }
+        mp_printf(print, ", ext_pin=%b", is_ext_pin(self));
```

Comment:
I think this line should be wrapped in `#if MICROPY_HW_PIN_EXT_COUNT` / `#endif`, so that on boards that do not have any ext pins (eg PICO) the `ext_pin` part is never printed.

Also, is `ext_pin` even needed to be printed?  This is not a keyword argument accepted by the `Pin()` constructor. The name of the pin (starts with `EXT_`) is enough to indicate it's external.  Furthermore, `%b` uses lowercase true/false, whereas it should print uppercase to match the Python True/False objects.

Reply:
Removed.